radar-upload-backend/src/main/java/org/radarbase/upload/resource/RecordResource.kt

```diff
@@ -148,27 +148,19 @@ class RecordResource {
             @PathParam("recordId") recordId: Long,
             @Context response: HttpServletResponse): StreamingOutput {
 
-        val (contentType, file) = recordRepository.readContentFile(recordId, fileName)
+        val recordContent = recordRepository.readContent(recordId, fileName)
+                ?: throw NotFoundException("Cannot find content with record-id $recordId and file-name $fileName")
+
         response.status = Response.Status.OK.statusCode
-        response.setHeader("Content-type", contentType)
-        response.setHeader("Content-Disposition", "attachment, filename=$fileName")
-        val inputStream = file.binaryStream
+        response.setHeader("Content-type", recordContent.contentType)
+        response.setHeader("Content-Length", recordContent.content.length().toString())
+        response.setHeader("Last-Modified", recordContent.createdDate.toString())
+        val inputStream = recordContent.content.binaryStream
         return StreamingOutput{
-
-            val buffer = ByteArray(1024)
-            var length: Int
-            reader@ while (true)  {
-                length = inputStream.read(buffer)
-                if(length != -1) {
-                    it.write(buffer, 0, length)
-                } else {
-                    break@reader
-                }
+            it.use {
```

Comment:
The output stream should not be closed, see https://stackoverflow.com/a/39575879/574082, so `use` should not be used on the outputStream. For the inputStream, `use` is fine.